metric-example/metric-expressions-for-k8s.md

```diff
@@ -0,0 +1,206 @@
+```:filter(and(in("dt.entity.cloud_application",entitySelector("type(cloud_application),entityId(~"CLOUD_APPLICATION-A26E32FC302257AB~")"))))```
+
+To adapt this to your scenario, simply replace "CLOUD_APPLICATION-A26E32FC302257AB" with the workload id of your workload. Again, you can find the id within the URL in the adress bar of your browser while viewing the workload in our Web UI.
+
+### Not all pods running
+Using the following query, you can find how many pods are not running compared to the number of desired pods of this workload.
+```
+( 
+  (builtin:cloud.kubernetes.workload.desiredPods):splitBy("dt.entity.cloud_application")
+  -(builtin:cloud.kubernetes.workload.pods):splitBy("dt.entity.cloud_application")
+)
+:filter(and(in("dt.entity.cloud_application",entitySelector("type(cloud_application),entityId(~"CLOUD_APPLICATION-A26E32FC302257AB~")"))))
+```
+
+### Not all containers running
+Using the following query, you can find how many containers are not running compared to the number of desired containers of this workload.
+```
+( 
+  (builtin:cloud.kubernetes.pod.desiredContainers):parents:splitBy("dt.entity.cloud_application")
+  -(builtin:cloud.kubernetes.pod.containers):parents:splitBy("dt.entity.cloud_application")
+)
+:filter(and(in("dt.entity.cloud_application",entitySelector("type(cloud_application),entityId(~"CLOUD_APPLICATION-A26E32FC302257AB~")"))))
+```
+
+## Workloads resource utilization and optimization
+Let's use some metric expressions to understand if we have **proper requests and limits** in place for our workloads. 
+We start by looking into the relation between usage and requests. Usually, one tries to have usage just below reqeuest. So which workloads are requesting far more CPU or Memory, than they actually use - this is usually refered to as *slack*. The following examples will only walk you through CPU related topics. They can easily by adapted to memory, by replacing the CPU metrics with their memory counterpart metrics. 
+
+### Slack
+```
+(
+    (builtin:cloud.kubernetes.pod.cpuRequests:parents:splitBy("dt.entity.cloud_application_instance","dt.entity.cloud_application"):avg)
+    -(builtin:containers.cpu.usageMilliCores:parents:parents:splitBy("dt.entity.cloud_application_instance","dt.entity.cloud_application"):avg)
+)
+:splitBy("dt.entity.cloud_application"):avg
+:filter(and(in("dt.entity.cloud_application",entitySelector("type(cloud_application),entityId(~"CLOUD_APPLICATION-A26E32FC302257AB~")"))))
+```
+*Note: We are using **:parents**, to be able to split and filter by entities being higher in the Kubernetes entity hierarchy, which is defined as follows: container -> pod -> workload -> namespace -> cluster.
+You can also increase the scope of this query to multiple workloads, by expanding the scope of the filter or just removing it. Just be aware, that in very large environments, this can put a lot of stress onto your Dynatrace environment.
```

Comment:
```suggestion
You can also increase the scope of this query to multiple workloads by expanding the scope of the filter or just removing it. Just be aware, that in very large environments this can put a lot of stress on your Dynatrace environment.
```